backend/src/packages/auth/jwt.service.ts

```diff
@@ -0,0 +1,17 @@
+import { SignJWT } from 'jose';
+
+class JWTService {
+  private secret: Buffer;
+
+  public constructor(secret: string) {
+    this.secret = Buffer.from(secret, 'utf8');
+  }
+
+  public async signJWT(payload: Record<string, number>): Promise<string> {
+    return await new SignJWT(payload)
+      .setProtectedHeader({ alg: 'HS256' })
```

Comment:
Let's use alg as secret one

Reply:
You mean switching for HS to RS?

Reply:
> Provide as ALG: 'HS256' in base config but with default value

Will that be good enough? 

<img width="501" alt="Screenshot 2023-08-22 at 12 26 29" src="https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/b4aaad36-786f-4a8f-8c91-4c674281620c">

Reply:
No, as a separate declaration, so AUTH contains JWT_SECRET and AIG

Reply:
> No, as a separate declaration, so AUTH contains JWT_SECRET and AIG

You think that will be better? 
<img width="498" alt="Screenshot 2023-08-22 at 12 32 40" src="https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/a9528284-32e0-4821-a2c1-c6fcf8c74a53">
<img width="229" alt="Screenshot 2023-08-22 at 12 32 53" src="https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/cfef5e3f-be99-430c-b4a6-be6b4d3af1d7">

Reply:
Will rename JWT_ALGORITHM into JWT_ALG ofcourse

Reply:
> No, as a separate declaration, so AUTH contains JWT_SECRET and AIG

I hope that will do: 

<img width="237" alt="Screenshot 2023-08-22 at 12 55 21" src="https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/78289512-12de-4120-83f6-2beadf2ceaac">

<img width="480" alt="Screenshot 2023-08-22 at 12 55 46" src="https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/3069ae72-63d9-49b4-820d-b3dcffc16523">
<img width="235" alt="Screenshot 2023-08-22 at 12 55 56" src="https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/9c543c90-e48f-42e1-b488-03e2bd75a7d5">

Reply:
It would be
`
JWT: {
    SECRET_KEY: {//},
    ALG: {//}
}
`

Reply:
And ALG with default value so .env and .env.example don't contain ALG record

Reply:
> And ALG with default value so .env and .env.example won't contain ALG record

Okay, got it, hope it'll work: 

<img width="490" alt="Screenshot 2023-08-22 at 13 08 58" src="https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/b69be081-01c7-4838-8dd1-30bd85e11999">

Reply:
The last thing to replace `alg` prop in `ALG` object to `default`

Reply:
Understood, just to confirm: 

<img width="488" alt="Screenshot 2023-08-22 at 13 15 42" src="https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/2e22bdd9-c81c-4e96-8436-fb76f26a72f8">
<img width="258" alt="Screenshot 2023-08-22 at 13 15 52" src="https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/62558753/7e891a8c-874f-46ca-85ac-d2977f50f6b9">

Reply:
Also don't forget to import ALG to JWTService)

Reply:
can we just import it directly, or it is better to pass alg through constructor like a secret property?